homeassistant/components/deconz/logbook.py

```diff
@@ -131,9 +131,10 @@ def async_describe_events(
     @callback
     def async_describe_deconz_alarm_event(event: Event) -> dict:
         """Describe deCONZ logbook alarm event."""
-        deconz_alarm_event: DeconzAlarmEvent | None = _get_deconz_event_from_device_id(
-            hass, event.data[ATTR_DEVICE_ID]
+        deconz_alarm_event: DeconzAlarmEvent | DeconzEvent | None = (
+            _get_deconz_event_from_device_id(hass, event.data[ATTR_DEVICE_ID])
         )
+        assert deconz_alarm_event
```

Comment:
Given the other comments in the other typing PRs about avoiding asserts: Do you have thoughts about another way to handle this? One thought is to always return a non-None object from `_get_deconz_event_from_device_id` and raise `InvalidDeviceAutomationConfig` when not found, like in some of the other cases.

Reply:
You're further ahead than me :), this is a good suggestion!

Reply:
I redid it according to your suggestion